src/data/seq/seq.lean

```diff
@@ -45,6 +45,15 @@ def cons (a : α) : seq α → seq α
 /-- Get the nth element of a sequence (if it exists) -/
 def nth : seq α → ℕ → option α := subtype.val
 
+@[simp] lemma nth_cons_zero (s : seq α) (x : α) : nth (cons x s) 0 = some x :=
+by cases s; refl
+
+@[simp] lemma nth_cons_succ (s : seq α) (x : α) (n : ℕ) : nth (cons x s) (n + 1) = nth s n :=
+by cases s; refl
```

Comment:
Here's a few extra lemmas and some reformatting/argument reordering.

```suggestion
@[simp] theorem nth_mk (f hf) : @nth α ⟨f, hf⟩ = f := rfl

@[simp] theorem nth_nil (n : ℕ) : (@nil α).nth n = none := rfl
@[simp] theorem nth_cons_zero (a : α) (s : seq α) : (cons a s).nth 0 = a := rfl
@[simp] theorem nth_cons_succ (a : α) (s : seq α) (n : ℕ) : (cons a s).nth (n + 1) = s.nth n := rfl
```
Take of this what you want.